en/docs/debugging/debug.md

```diff
@@ -0,0 +1,258 @@
+# Debug Ballerina Programs
+
+When writing large-scale applications with complex logic, it is quite possible to have syntax, semantic,
+or runtime issues in your code. The Ballerina compiler comes in handy when detecting syntax and semantic issues.
+However, it is impossible for a compiler to detect runtime errors like logical errors because they occur during the
+program execution after a successful compilation. This is where the dedicated debugging tooling support becomes important.
+
+Currently, a feature-rich debugging experience for troubleshooting Ballerina applications
+is provided via the <a href="https://marketplace.visualstudio.com/items?itemName=WSO2.Ballerina" target="_blank">Ballerina Visual Studio Code extension</a>.
+
+## Start a debug session
+
+The Ballerina extension provides multiple options to debug Ballerina applications and the most convenient way will be using the context-aware debug CodeLens support.
+
+However, if you require launching debug sessions with additional configurations (i.e., program arguments, environment variables), you can use the `launch.json` configurations.
+>**Info:** For more information, see [Start a debug session with configurations](#start-a-debug-session-with-configurations)).
+
+### Start a program debug session
+
+Follow the steps below to start a quick debug session using CodeLens.
+
+1. Open the folder, which includes the Ballerina program you want to debug and open the source file in the editor.
+
+2. Add the debug points you require by clicking in front of the line numbers of the file you want to debug.
+
+3. Click the `Debug` CodeLens, which will appear just above the `main()` method.
+
+![Start_Main Quick Debug Session](/img/start-quick-main-debug-session.gif)
+
+<br/>
+
+### Start a test debug session
+
+The Ballerina test functions can also be debugged using CodeLens. The `debug` CodeLens will automatically appear on top of each Ballerina test function
+and you are able to execute/debug only the selected test case by clicking on the corresponding CodeLens as shown below.
+
+![Start_Test Quick Debug Session](/img/start-quick-test-debug-session.gif)
+
+<br/>
+
+### Start a debug session with configurations
+
+Follow the steps below to start a debug session with configurations. All the configurations need to be added in the `launch.json` file.
+
+1. Open the folder, which includes the Ballerina program you want to debug and select the file.
+
+2. Press the **Control + Shift + D** keys (for Mac: **Command + Shift +D**) to launch the Debugger view.
+
+3. Click **create a launch.json** file and then select **Ballerina Debug** as the **Environment**.
+
+   You view the opened `launch.json` file.
```

Comment:
```suggestion
   You can view the opened `launch.json` file.
```

Reply:
added with [Address the latest review suggestions](https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/pull/20/commits/71d66faeecce06a07d7eccec8edf1b3c3eb42ffb)